chipsec/chipset.py

```diff
@@ -739,19 +763,28 @@ def is_register_defined(self, reg_name):
             #raise RegisterNotFoundError, ('RegisterNotFound: %s' % reg_name)
             return False
 
-    def get_register_def(self, reg_name):
+    def get_register_def(self, reg_name, bus_index=0):
         reg_def = self.Cfg.REGISTERS[reg_name]
-        if reg_def["type"] == "pcicfg" and "device" in reg_def:
+        if (reg_def["type"] == "pcicfg" or reg_def["type"] == "mmcfg") and ("device" in reg_def):
             dev_name = reg_def["device"]
             if dev_name in self.Cfg.CONFIG_PCI:
                 dev = self.Cfg.CONFIG_PCI[dev_name]
                 reg_def['bus'] = dev['bus']
                 reg_def['dev'] = dev['dev']
                 reg_def['fun'] = dev['fun']
+                if dev_name in self.Cfg.BUS:
+                    reg_def['bus'] = self.Cfg.BUS[dev_name][bus_index]
```

Comment:
Is there a way to ensure that bus_index is a valid index?

Reply:
Calling get_register_bus() or get_device_bus() will return the valid list.

Reply:
what if bus index is not in the range of `self.Cfg.BUS[dev_name][bus_index]`.  Please add a check to avoid an exception.